algebird-core/src/main/scala/com/twitter/algebird/HyperLogLog.scala

```diff
@@ -231,6 +261,34 @@ object HyperLogLog {
     require(err >= 0.00003 && err < 1.0, s"Error must be in (0.00003, 1.0): $err")
     math.ceil(2.0 * math.log(1.04 / err) / math.log(2.0)).toInt
   }
+
+  def initialEstimate(bits: Int, size: Int, zeroCnt: Int, z: Double): Double = {
+    val sizeDouble: Double = size.toDouble
+    val smallE = 5 * sizeDouble / 2.0
+    val factor = alpha(bits) * sizeDouble * sizeDouble
+
+    val e: Double = factor * z
+
+    // There are large and small value corrections from the paper We
+    // stopped using the large value corrections since when using
+    // Long there were pathologically bad results.
+    //
+    // See https://github.com/twitter/algebird/issues/284
+    if (e > smallE || zeroCnt == 0) e
+    else size * scala.math.log(size.toDouble / zeroCnt)
+  }
+
+  def asApprox(bits: Int, v: Double): Approximate[Long] = {
+    val stdev = 1.04 / scala.math.sqrt(twopow(bits))
+    val lowerBound = math.floor(math.max(v * (1.0 - 3 * stdev), 0.0)).toLong
+    val upperBound = math.ceil(v * (1.0 + 3 * stdev)).toLong
```

Comment:
actually I guess the `floor` and `ceil` should be reversed. We can't widen the bound and still claim the probability is the same. I think this is a (minor) off-by-one bug.